onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -490,7 +496,14 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<onnxruntime:
 
       // Run TRT inference
       std::lock_guard<OrtMutex> lock(*(trt_state->tensorrt_mu_ptr));
-      trt_state->context->enqueue(batch_size, &buffers[0], nullptr, nullptr);
+      bool ret = trt_state->context->enqueue(batch_size, &buffers[0], nullptr, nullptr);
+      if (!ret) {
+        if (trt_state->context->getEngine().getMaxBatchSize() < batch_size) {
+	  return common::Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT,
```

Comment:
It might be better to pass max_batch_size_ to compute_func as an argument and compare it to batch_size directly since getMaxBatchSize() actually just returns what we set in ORT_TENSORRT_MAX_BATCH_SIZE. Also it will eliminate the function call in compute_func, which will cost extra time during inference

Reply:
I take it back. The function call only happened when enqueue failed, so it should be fine to keep it in compute_func

Reply:
Yes, I purposely only check in the failure case